editor.planx.uk/src/pages/FlowEditor/components/Sidebar/index.tsx

```diff
@@ -152,6 +191,88 @@ const DebugConsole = () => {
   );
 };
 
+const LinkComponent = (props: {
+  primaryColour?: string;
+  titleIcon?: string | SvgIconProps;
+  title: string;
+  link: string;
+  description?: string;
+  isPublished?: boolean;
+}) => {
+  const [copyMessage, setCopyMessage] = useState<"copy" | "copied">("copy");
+
+  return (
+    <Box
+      sx={{
+        opacity:
+          props.isPublished || props.isPublished === undefined ? "100%" : "50%",
+      }}
+      display={"flex"}
+      flexDirection={"column"}
+      gap={"8px"}
+      mb={1}
+    >
+      <Box
+        display={"flex"}
+        flexDirection={"row"}
+        alignItems={"center"}
+        gap={"7px"}
+      >
+        {typeof props.titleIcon === "string" ? (
```

Comment:
This took me a minute to parse and to understand that `titleIcon` is either a URL to a logo, `undefined` or an SVG icon.

It may be a better design to make the props for this component a discriminated union of two types - one for a logo, and one for an icon. This would mean that as opposed to multiple optional properties without any relationship with each other, we can have a stricter type which better represents how this component can get used.

Here's an example - 

```ts
// Shared values which are always present
interface BaseProps { 
  title: string;
  // ...
}

// Props always required if we're representing a logo
interface LogoProps extends BaseProps {
  type: "logo";
  primaryColour: string;
  logoURL: string;
}

// Props always required if we're representing an icon
interface IconProps extends BaseProps {
  type: "icon";
  icon: SvgIconProps;
}

// Our union type
type Props =  LogoProps | IconProps;


// Examples
const validLogo: Props = {
  title: "abc",
  type: "logo",
  primaryColour: "#FFF",
  logoURL: "abc"
}

const validIcon: Props = {
  title: "abc",
  type: "icon",
  icon: <SomeIcon/>
}

const invalidType: Props = {
  title: "abc",
  type: "logo",
  icon: <SomeIcon/>
}
```

Reply:
This may seem a little complex initially, but it's a super useful pattern to learn - we're trading a little bit of type complexity upfront for a lot of type safety, a better developer experience, and clearer code (imho!) 👌